apps/api/src/server/utils/__tests__/migration-dates.test.js

```diff
@@ -0,0 +1,60 @@
+import { changeEndTime, handleDateTimeToUTC } from '#utils/migration-dates.js';
+
+describe('odw-date', () => {
+	describe('changeEndTime', () => {
+		describe('if time is 00:00', () => {
+			test('should change time to 23:59', () => {
+				const date = new Date('2022-10-05T00:00:00');
+				changeEndTime(date);
+				expect(date.getHours()).toBe(23);
+				expect(date.getMinutes()).toBe(59);
+			});
+		});
+		describe('if time is not 00:00', () => {
+			test('should not change time', () => {
+				const date = new Date('2022-10-05T12:30:00');
+				changeEndTime(date);
+				expect(date.getHours()).toBe(12);
+				expect(date.getMinutes()).toBe(30);
+			});
+		});
+	});
+
+	describe('handleDateTimeToUTC', () => {
+		describe('Invalid Inputs', () => {
+			test('should return null for an empty datetimeString', () => {
+				const result = handleDateTimeToUTC('', { isEndDate: false });
+				expect(result).toBeNull();
+			});
+
+			test('should return null for an invalid datetimeString', () => {
+				const result = handleDateTimeToUTC('invalid date', { isEndDate: false });
+				expect(result).toBeNull();
+			});
+
+			describe('when shouldSeparateTime is true', () => {
```

Comment:
For clarify can we say when "isEndDate is false" as for a second I thought "shouldSeparateTime" was another paramter.

Reply:
fixed!